napari/_vispy/canvas.py

```diff
@@ -382,9 +386,10 @@ def _process_mouse_event(
             return
 
         # Add the view ray to the event
-        event.view_direction = self.viewer.camera.calculate_nd_view_direction(
-            self.viewer.dims.ndim, self.viewer.dims.displayed
-        )
+        # event.view_direction = self.viewer.camera.calculate_nd_view_direction(
+        #     self.viewer.dims.ndim, self.viewer.dims.displayed
+        # )
+        event.view_direction = self._calculate_view_direction(event.pos)
         event.up_direction = self.viewer.camera.calculate_nd_up_direction(
             self.viewer.dims.ndim, self.viewer.dims.displayed
         )
```

Comment:
Is it possible that this method also needs to be updated similarly?

Reply:
I am not familiar with this fucntion, it seems have something to do with the switch between 2D and 3D mode.